Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Controller.mo

```diff
@@ -0,0 +1,2498 @@
+    final nout=nChi)
+    if not have_fixSpeConWatPum
+    "Replicate design condenser water temperature"
+    annotation (Placement(transformation(extent={{540,190},{560,210}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.MultiMin mulMin(
+    final nin=nChi)
+    if not have_fixSpeConWatPum
+    annotation (Placement(transformation(extent={{-460,150},{-440,170}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.Switch chiMinFloSet
+    "Chiller water minimum flow setpoint"
+    annotation (Placement(transformation(extent={{480,110},{500,130}})));
+
+  Buildings.Controls.OBC.CDL.Routing.BooleanScalarReplicator uChiSwi(
+    final nout=nChi)
+    "In chiller stage up process"
+    annotation (Placement(transformation(extent={{460,340},{480,360}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Switch uChiStaPro[nChi]
+    "Chiller head pressure control status"
+    annotation (Placement(transformation(extent={{640,340},{660,360}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Pre preConPumLeaSta
+    "Lead condenser water pump status from previous step"
+    annotation (Placement(transformation(extent={{480,-260},{500,-240}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Generic.EquipmentRotation.ControllerTwo equRot if have_heaChiWatPum
+    "Rotates two pumps or groups of pumps"
+    annotation (Placement(transformation(extent={{260,570},{280,590}})));
+
+  Buildings.Controls.OBC.CDL.Integers.Sources.Constant conInt1[nChiWatPum](
+    final k={1,2}) if have_heaChiWatPum
+    "Two pumps or groups of pumps only"
+    annotation (Placement(transformation(extent={{320,600},{340,620}})));
+
+  Buildings.Controls.OBC.CDL.Integers.Sources.Constant conInt2[nChiWatPum](
+    final k={2,1}) if have_heaChiWatPum
+    "Two pumps or groups of pumps only"
+    annotation (Placement(transformation(extent={{320,550},{340,570}})));
+
+  Buildings.Controls.OBC.CDL.Integers.Switch intSwi[2] if have_heaChiWatPum
+    "Two devices or groups of devices"
+    annotation (Placement(transformation(extent={{380,570},{400,590}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Latch chiStaUp
+    "In chiller stage up process"
+    annotation (Placement(transformation(extent={{380,310},{400,330}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Pre pre2 "Stage cooling tower"
+    annotation (Placement(transformation(extent={{80,-450},{100,-430}})));
+
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToInteger booToInt[nChiWatPum] if have_heaChiWatPum
+    "Convert boolean to integer"
+    annotation (Placement(transformation(extent={{580,550},{600,570}})));
+
+  Buildings.Controls.OBC.CDL.Integers.MultiSum totChiPum(
+    final nin=nChiWatPum) if have_heaChiWatPum
+    "Total enabled chilled water pump"
+    annotation (Placement(transformation(extent={{620,550},{640,570}})));
+
+  Buildings.Controls.OBC.CDL.Integers.GreaterThreshold intGreThr(
+    final t=1)
+    "Check if more than one pump is enabled, if yes, then it means lag pump is enabled"
```

Comment:
BUG: This instance is not conditional. However, its input is only connected with
`connect(totChiPum.y, intGreThr.u)` where `totChiPum` is a conditional instance depending on `have_heaChiWatPum`. This yields a singular system if `not have_heaChiWatPum`.